src/creatures/creature.cpp

```diff
@@ -813,7 +813,17 @@ bool Creature::dropCorpse(std::shared_ptr<Creature> lastHitCreature, std::shared
 					player->sendLootMessage(lootMessage.str());
 				}
 
-				if (player->checkAutoLoot(monster->isRewardBoss()) && corpseContainer && mostDamageCreature->getPlayer()) {
+				auto &fromPos = player->getPosition();
+				auto &toPos = corpse->getPosition();
+				stdext::arraylist<Direction> dirList(128);
+				FindPathParams fpp;
+				fpp.minTargetDist = 0;
+				fpp.maxTargetDist = 1;
+				fpp.fullPathSearch = true;
+				fpp.clearSight = true;
+				fpp.maxSearchDist = 0;
+
+				if (player->checkAutoLoot(monster->isRewardBoss()) && corpseContainer && mostDamageCreature->getPlayer() && g_game().map.getPathMatching(fromPos, dirList, FrozenPathingConditionCall(toPos), fpp) ) {
```

Comment:
isn't better if you use `player->getPosition()` and `corpse->getPosition()` direct, instead of create variables?

Reply:
liked the tip, didnt want to insert the methods directly inside the if to not extend the line too much.